rcl_lifecycle/src/rcl_lifecycle.c

```diff
@@ -161,17 +161,21 @@ rcl_lifecycle_transition_fini(
 
   rcl_ret_t ret = RCL_RET_OK;
 
-  if (rcl_lifecycle_state_fini(transition->start, allocator) != RCL_RET_OK) {
-    ret = RCL_RET_ERROR;
+  if (transition->start) {
```

Comment:
why is that needed? Calling `fini/deallocate` on a `nullptr` should be totally fine.

Reply:
After checking codes, Yes.